crates/router/src/lib.rs

```diff
@@ -130,24 +161,25 @@ fn calculate_default_headers<'a>(
         .context("expected 'host' header to be UTF-8 encoded but it was not")?
         .unwrap_or_else(|| "localhost".to_owned());
 
-    let matched_route =
-        spin_http::routes::RoutePattern::sanitize_with_base(base.clone(), raw_route.clone());
-
-    let path_info = spin_http::routes::RoutePattern::from(base.clone(), raw_route.clone())
-        .relative(&abs_path)?;
     let full_url = format!("{}://{}{}", scheme, host, abs_path);
-    let component_route = raw_route
-        .strip_suffix("/...")
-        .unwrap_or(&raw_route)
-        .to_owned();
-
-    res.push((PATH_INFO, path_info));
-    res.push((FULL_URL, full_url));
-    res.push((MATCHED_ROUTE, matched_route));
-    res.push((BASE_PATH, base));
-    res.push((RAW_COMPONENT_ROUTE, raw_route));
-    res.push((COMPONENT_ROUTE, component_route));
-    res.push((CLIENT_ADDR, "127.0.0.1:0".to_owned()));
+
+    res.push((owned_path_info, path_info));
+    res.push((owned_full_url, full_url));
+    res.push((owned_matched_route, route_match.based_route().to_string()));
+
+    res.push((owned_base_path, base.to_string()));
+    res.push((
+        owned_raw_component_route,
+        route_match.raw_route().to_string(),
+    ));
+    res.push((owned_component_route, route_match.raw_route_or_prefix()));
+    res.push((owned_client_addr, "127.0.0.1:0".to_owned()));
+
+    for (wild_name, wild_value) in route_match.named_wildcards() {
+        let wild_header = format!("SPIN_PATH_MATCH_{}", wild_name.to_ascii_uppercase()); // TODO: safer
```

Comment:
I'm not sure what the TODO is here. Perhaps this can be expanded on?

Reply:
Upstream copy artifact, removed.